sympy/solvers/ode/single.py

```diff
@@ -2615,5 +2615,196 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
+            _coeff2 = r[c3].match(a - b)
+            if _coeff2 is None:
+                return False
+            # matching with standerd form for c3
+            coeff2 = factor(_coeff2[a]).match(c4**2*(x)**(2*a4))
+            if coeff2 is None:
+                return False
+
+            if _coeff2[b] == 0:
+                coeff2[d4] = 0
+            else:
+                coeff2[d4] = factor(_coeff2[b]).match(d4**2)[d4]
+
+            self.rn = {'n':coeff2[d4], 'a4':coeff2[c4], 'd4':coeff2[a4]}
+            self.rn['c4'] = coeff1[a4]
+            self.rn['b4'] = point
+            return True
+        return False
+
+    def _get_general_solution(self, *, simplify_flag: bool = True):
+        f = self.ode_problem.func.func
+        x = self.ode_problem.sym
+        n = self.rn['n']
+        a4 = self.rn['a4']
+        c4 = self.rn['c4']
+        d4 = self.rn['d4']
+        b4 = self.rn['b4']
+        n = sqrt(n**2 + Rational(1, 4)*(c4 - 1)**2)
+        (C0, C1) = self.ode_problem.get_numbered_constants(num=2)
+        return [Eq(f(x), ((x**(Rational(1-c4,2)))*(C0*besselj(n/d4,a4*x**d4/d4)
+            + C1*bessely(n/d4,a4*x**d4/d4))).subs(x, x-b4))]
+
+
+class LinearAiry2nd(SingleODESolver):
+    r"""
+    Gives solution of the Airy differential equation
+
+    .. math :: \frac{d^2y}{dx^2} + (a + b x) y(x) = 0
+
+    in terms of Airy special functions airyai and airybi.
+
+    Examples
+    ========
+
+    >>> from sympy import dsolve, Function
+    >>> from sympy.abc import x
+    >>> f = Function("f")
+    >>> eq = f(x).diff(x, 2) - x*f(x)
+    >>> dsolve(eq)
+    Eq(f(x), C1*airyai(x) + C2*airybi(x))
+    """
+    hint = "2nd_linear_airy"
+    has_integral = False
+
+    def _matches(self):
+        eq = self.ode_problem.eq_high_order_free
+        f = self.ode_problem.func
+        order = self.ode_problem.order
+        x = self.ode_problem.sym
+        df = f.diff(x)
+        a3 = Wild('a3', exclude=[f, df, f.diff(x, 2)])
+        b3 = Wild('b3', exclude=[f, df, f.diff(x, 2)])
+        c3 = Wild('c3', exclude=[f, df, f.diff(x, 2)])
+        a4 = Wild('a4', exclude=[x,f,df])
+        b4 = Wild('b4', exclude=[x,f,df])
+        does_match = False
+        deq = a3*(f.diff(x, 2)) + b3*df + c3*f
+        r = collect(eq,
+            [f.diff(x, 2), df, f]).match(deq)
+        if r:
+            if not all([r[key].is_polynomial() for key in r]):
+                n, d = eq.as_numer_denom()
+                eq = expand(n)
+                r = collect(eq,
+                    [f.diff(x, 2), f.diff(x), f]).match(deq)
+        if order==2 and r and r[a3] != 0:
+            p = cancel(r[b3]/r[a3])  # Used below
+            q = cancel(r[c3]/r[a3])
+            if p.is_zero:
+                self.rn = q.match(a4+b4*x)
+                if self.rn and self.rn[b4] != 0:
+                    self.rn = {'b':self.rn[a4],'m':self.rn[b4]}
+                    does_match = True
+        return does_match
+
```

Comment:
@oscarbenjamin According to my understanding about `2nd_linear_airy`, I attempted to simplify the match function something like :
```python
    def _matches(self):
        eq = self.ode_problem.eq_high_order_free
        f = self.ode_problem.func
        order = self.ode_problem.order
        x = self.ode_problem.sym
        df = f.diff(x)
        a4 = Wild('a4', exclude=[x,f,df])
        b4 = Wild('b4', exclude=[x,f,df])
        match = self.ode_problem.get_linear_coefficients(eq, f, order)
        does_match = False
        if order == 2 and match and match[2] != 0:
            if match[1].is_zero:
                self.rn = match[0].match(a4+b4*x)
                if self.rn and self.rn[b4] != 0:
                    self.rn = {'b':self.rn[a4],'m':self.rn[b4]}
                    does_match = True
        return does_match
```
It passes all the tests but wanted to ask you does this diff look ok?

Reply:
Does `get_linear_coefficients` divide by the leading coefficient?

What happens if you have something like `x y'' + x y = 0`?

Reply:
> x y'' + x y = 0

It fails for this. I thought `self.ode_problem.eq_high_order_free` passing in the `get_linear_coefficients` would work as eq_high_order_free removes leading coefficient but it only remove `f(x)` not x.

Reply:
if we replace `self.rn = match[0].match(a4+b4*x)` with `self.rn = cancel(match[0]/match[2]).match(a4+b4*x)` , It will solve the problem :thinking: